UniversityApp.DataAccess/Repositories/StudentsRepository.cs

```diff
@@ -25,19 +26,30 @@ public async Task<List<Student>> GetStudents()
             return result;
         }
 
-        public async Task<(List<Student> Students, int TotalRecordCount)> GetStudents(List<Expression<Func<Student, bool>>> predicates, int skip, int amount)
+        public async Task<(List<Student> Students, int TotalRecordCount)> GetStudents(StudentSearchParameters studentSearchParameters)
         {
             var students = _context.Students.AsQueryable();
 
-            if(predicates != null)
+            if (!String.IsNullOrEmpty(studentSearchParameters.FirstName))
             {
-                foreach(var predicate in predicates)
-                {
-                    students = students.Where(predicate);
-                }
+                students = students.Where(s => s.FirstName.Contains(studentSearchParameters.FirstName));
             }
+            if (!String.IsNullOrEmpty(studentSearchParameters.LastName))
+            {
+                students = students.Where(s => s.LastName.Contains(studentSearchParameters.LastName));
+            }
+            if (studentSearchParameters.Age != null)
+            {
+                students = students.Where(s => s.Age == studentSearchParameters.Age);
+            }
+            if (!String.IsNullOrEmpty(studentSearchParameters.Gender))
+            {
+                students = students.Where(s => s.Gender.Contains(studentSearchParameters.Gender));
+            }
+
+            int skip = (studentSearchParameters.PageNumber - 1) * studentSearchParameters.PageSize;
```

Comment:
name 'skip' doesn't mean anything specific. Plese use different naming that indicates number of results that should be skipped